File: python/primeFactorize.py
def prime_factorize(N):
    N2 = N
    ans = []
    i = 2
    while i * i <= N:
        cnt = 0
        while N2 % i == 0:
            N2 //= i
            cnt += 1

        if cnt > 0:
            ans.append([i, cnt])
        i += 1

    if N2 != 1:
        ans.append([N2, 1])

    return ans

File: rust/prime_factorize.rs
fn prime_factorize(n: u64) -> Vec<(u64, usize)> {
    let mut i = 2;
    let mut ansvec = vec![];
    let mut n2 = n;
    while i * i <= n {
        let mut b = 0;
        while n2 % i == 0 {
            n2 /= i;
            b += 1;
        }
        if b > 0 {
            ansvec.push((i, b));
        }
        i += 1;
    }
    if n2 > 1 {
        ansvec.push((n2, 1));
    }
    ansvec
}
